base/applications/osk/main.c

```diff
@@ -365,6 +397,19 @@ BOOL OSK_DlgCommand(WPARAM wCommand, HWND hWndControl)
         bExtendedKey = FALSE;
     ScanCode &= 0xFF;
 
+    /*
+        The user could've pushed one of the key buttons of the dialog that
+        can trigger particular function toggling (Caps Lock, Num Lock or Scroll Lock). Update
+        (invalidate) the LED icon resources accordingly.
+    */
+    for (i = 0; i < _countof(LedKey); i++)
+    {
+        if (ScanCode & 0x013A || ScanCode & 0x0145 || ScanCode & 0x0146)
```

Comment:
This is a very weird check, especially since a few lines above is this: `ScanCode &= 0xFF;`

Reply:
What exactly do you find weird on this? The check here merely checks for the respective scan code values, each of those being one of the keys (Scroll Lock, Num Lock and Caps Lock). 

The resource buttons for the keys are defined with these scan codes and when the user clicks one of these buttons, `OSK_DlgCommand()` receives `wCommand` data. This is better than invalidating the LED resources for without a reason.

Reply:
1. Did you mean `ScanCode == 0x013A` and etc? (because these values don't look like bit masks)
2. Do you realize `ScanCode &= 0xFF;` operation will leave only least 8 bits in the variable? So this condition will always be false.

Reply:
This triggers for a lot of unrelated keys, is that intentional?

Reply:
Yeah this part of code still looks strange. @Fraizeraust did you mean `if (LedKey[i].vKey == ScanCode)`?

Reply:
@binarymaster These are just scan codes so doing `if (LedKey[i].vKey == ScanCode)` is incorrect because the condition will always be false (virtual keys have different values). But yes, doing `ScanCode & 0x013A` is also incorrect because these are not bit masks. 

I'll fix as well as simplify the check (adding an extra member to the structure, *wScanCode*, and fill it with scan code values data on the array).

Reply:
How about https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-mapvirtualkeyw ?